lib/SILGen/SILGenPattern.cpp

```diff
@@ -2437,64 +2439,69 @@ void PatternMatchEmission::emitSharedCaseBlocks() {
       // Then emit the case body into the caseBB.
       SGF.B.setInsertionPoint(caseBB);
     }
-    
+
+    // Make sure that before/after we emit the body we have the same cleanups
+    // depth to ensure that all cleanups from inside the case body have been
+    // popped.
     assert(SGF.getCleanupsDepth() == PatternMatchStmtDepth);
+    SWIFT_DEFER { assert(SGF.getCleanupsDepth() == PatternMatchStmtDepth); };
+
+    // If we do not have any bound decls, just emit the case body... there is no
+    // further work to do.
+    if (!caseBlock->hasBoundDecls()) {
+      return emitCaseBody(caseBlock);
```

Comment:
This was why this originally failed. I should have used a continue instead of a return.